image/image.go

```diff
@@ -8,10 +8,24 @@ import (
 // Type must be a "Backing file format name string" that appears in QCOW2.
 type Type string
 
+type Extent struct {
+	// Offset from start of the image in bytes.
+	Start int64 `json:"start"`
+	// Length of this extent in bytes.
+	Length int64 `json:"length"`
+	// Set if this extent is allocated.
+	Allocated bool `json:"allocated"`
+	// Set if this extent is read as zeros.
+	Zero bool `json:"zero"`
+	// Set if this extent is compressed.
+	Compressed bool `json:"compressed"`
+}
+
 // Image implements [io.ReaderAt] and [io.Closer].
 type Image interface {
 	io.ReaderAt
 	io.Closer
+	Next(start, length int64) (Extent, error)
```

Comment:
Maybe just
```suggestion
	Extent(start, length int64) (Extent, error)
```

Reply:
Can work, but this does give a hint that this is useful for iterating over the image - similar to the tar package.
https://pkg.go.dev/archive/tar#Reader

Another option:

```go
NextExtent(start, length int64) (Extent, error)
```

But it repeats Extent in the name and in the return value.

Reply:
The difference is that tar.Reader.Next is stateful while Image.Next here seems stateless

Reply:
Good point, Next() is stateless. It exposes the structure of the image. It is more like `seek(SEE_HOLE)` or `seek(SEEK_DATA)`.

Lets go with Extent().

Reply:
Renamed.